### PR TITLE
OKTA-518743 : OV channel switch translation text fix implementation

### PR DESCRIPTION
**Pull request by glenfannin-okta**
## Description:
Purpose of this PR is to fix the OV channel switch button text. The current text has an anchor tag embedded, I had to create a new renderer to support translation strings that contain anchor tags and text combined.

## PR Checklist

- [x] Have you verified the basic functionality for this change?

- [x] Added unit tests?

- [x] Added e2e tests

- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?

- Did you verify the change by running downstream monolith artifacts? (YES | NO | UNSURE)

### Screenshot/Video:
image

### Issue:

- [OKTA-518743](https://oktainc.atlassian.net/browse/OKTA-518743)

### CODE REVIEW

File: src/v3/src/components/Button/Button.tsx [hunk 1]
```diff
@@ -64,6 +65,8 @@ const Button: UISchemaElementComponent<{
       {...(dataSe && { 'data-se': dataSe } )}
       // eslint-disable-next-line react/jsx-props-no-spreading
       {...(type !== 'submit' && { onClick })}
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      {...(ariaLabel && { 'arial-label': ariaLabel } )}
```
[the remark this answers is not in the document]
Comment (glenfannin-okta) [reply src/v3/src/components/Button/Button.tsx:69]: Not from what I have seen, in this specific case the OV "Can't scan" button explicitly had an aria-label. It is the only one I have seen defined in v2.

File: src/v3/src/components/TextWithLink/TextWithLink.tsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  uischema: TextWithLinkElement
+}> = ({ uischema }) => {
+  const {
+    content,
+    step,
+    stepToRender,
+    linkClass,
+    action: actionFn,
+  } = uischema.options;
+  const onSubmitHandler = useOnSubmit();
```
[the remark this answers is not in the document]
Comment (glenfannin-okta) [reply src/v3/src/components/TextWithLink/TextWithLink.tsx:33]: In v2, it just loads a new view into the DOM, however, with the changes made to v3 channel selection, it looks like a XHR request is made. The anchor tag is embedded in the translation string, discussed this with @lesterchoi-okta and I believe this is to make it functional, but we will request a new key for this.

[the remark this answers is not in the document]
Comment (shuowu-okta) [reply src/v3/src/components/TextWithLink/TextWithLink.tsx:33]: How about name it as `LinkButton`? Link means it looks like a link, Button means it behaves like a button (xhr). TextWithLink seems a component to open a new tab to me.

[the remark this answers is not in the document]
Comment (glenfannin-okta) [reply src/v3/src/components/TextWithLink/TextWithLink.tsx:33]: Are you recommending changing the component name to `TextWithLinkButton`? 
This the the translated string: `Or <a href="#" class="switch-channel-link">try a different way</a> to set up Okta Verify.`
I tried to be descriptive of the usecase with the component name.

File: src/v3/src/components/Button/Button.tsx (hunk 1 above)
Comment (shawyu-okta) [inline src/v3/src/components/Button/Button.tsx:69]:
```suggestion
      {...(ariaLabel && { 'aria-label': ariaLabel } )}
```

File: src/v3/src/components/TextWithLink/TextWithLink.tsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+    return () => {
+      element.removeEventListener?.('click', handleClick);
+    };
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  return (
+    <Box>
+      <Box dangerouslySetInnerHTML={{ __html: content }} />
```
Comment (shawyu-okta) [inline src/v3/src/components/TextWithLink/TextWithLink.tsx:57]: Based on this implementation I might be more inclined to call this component `TextWithHtml`

Comment (glenfannin-okta) [reply src/v3/src/components/TextWithLink/TextWithLink.tsx:57]: That makes perfect sense. 🙏  Thank you! Will update.

File: src/v3/src/components/Form/renderers.tsx [hunk 4]
```diff
@@ -106,6 +107,10 @@ export default [
     },
     renderer: PhoneAuthenticator,
   },
+  {
+    tester: ({ type }) => type === 'TextWithLink',
```
Comment (shawyu-okta) [inline src/v3/src/components/Form/renderers.tsx:111]: One more update needed here
```suggestion
    tester: ({ type }) => type === 'TextWithHtml',
```

Comment (glenfannin-okta) [reply src/v3/src/components/Form/renderers.tsx:111]: Oh yea, fixed this just running jest to update tests appropriately and will push.

**Review by shawyu-okta — APPROVED** [review]

**Pull request merged**